crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -346,19 +428,136 @@ impl InspectorStack {
                 if new_status != status ||
                     (new_status == InstructionResult::Revert && new_retdata != retdata)
                 {
-                    return (new_status, new_gas, new_retdata);
+                    Some((new_status, new_gas, new_retdata))
+                } else {
+                    None
                 }
-            }
+            },
+            self,
+            data
         );
-
         (status, remaining_gas, retdata)
     }
+
+    fn transact_inner<DB: DatabaseExt + DatabaseCommit>(
+        &mut self,
+        data: &mut EVMData<'_, DB>,
+        transact_to: TransactTo,
+        caller: Address,
+        input: Bytes,
+        gas_limit: u64,
+        value: U256,
+    ) -> (InstructionResult, Option<Address>, Gas, Bytes) {
```

Comment:
would love if we unit tested this with a simple regression test, but can do in follow up

Reply:
we don't really run any tests for isolation rn besides two I've added for selfdestruct/tstore, not sure how we can address this without running all tests for both modes

Reply:
Should we add a flag to run these on CI perhaps every 24hrs on `--isolate`? Something like that?

Reply:
yeah, that could work, we'd have to filter some tests out for such runs because we have some fixtures with hardcoded gas usage which doesn't match for `--isolate` and also that one failing sablier-v2 test